In [1]:
import sys
sys.path.append("..")
from pathlib import Path
import plotly.graph_objects as px
from utils.transcription_utils import get_transcription_info,get_stats,get_total_stats
import csv
import pandas as pd
import time
import tarfile
from tqdm import tqdm

In [2]:
save_path=Path("/media/dataset-harddisk/munikumar/utils/audio_utils/data/kathbath")

In [3]:
dataset_dir=Path("/media/dataset-harddisk/munikumar/hindi_dataset/kathbath/kathbath_raw")

In [4]:
extract_path=Path("/media/dataset-harddisk/munikumar/hindi_dataset/kathbath/kathbath_raw/extract")

In [5]:
def extract_files(extraction_path):
    for tar_file in extraction_path.parent.glob("*.tar"):
        start_time = time.time()
        with tarfile.open(tar_file) as tar:
            with tqdm(desc=f"Extracting {tar_file.name} ") as progress_bar:
                member = tar.next()
                while member is not None:
                    progress_bar.update(1)
                    if member.name.startswith("kb_data_clean_m4a/hindi/"):
                        tar.extract(member, path=str(extraction_path))
                    member = tar.next()
        end_time = time.time()
        execution_time = end_time - start_time
        print(f"{tar_file.name} extraction time: {execution_time:.4f} seconds")

In [6]:
# extract_files(extract_path)

In [7]:
def load_transcription(transcription_path):
    transcriptions={}
    with open(transcription_path, "r", encoding="utf-8") as fp:
        for transcription_data in fp.read().splitlines():
            audio_file = transcription_path.parent.name+"/audio/"+transcription_data.split("\t")[0]
            transcription = transcription_data.split("\t")[-1]
            transcriptions[audio_file] = transcription
    return transcriptions

In [8]:
dataset_path=extract_path/"kb_data_clean_m4a"/"hindi"

In [9]:
# test_transcription_data=load_transcription(dataset_path/"test"/"transcription_n2w.txt")
# testk_transcription_data=load_transcription(dataset_path/"test_known"/"transcription_n2w.txt")
# test_transcription_data.update(testk_transcription_data)
# get_transcription_info(test_transcription_data,dataset_path,save_path,"test")
get_stats(save_path/"test_non_hindi_transcripts.json")

**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 8:1:8
Loss duration : 0:0:41
Remaining duration : 8:0:26
Drop by percentage: 0.14

Total audio files : 5059
Files that would be drop: 6
Remaining audio files : 5053
Drop by percentage: 0.12
**************************************************
Non hindi tokes : ['ऍ', 'ऽ', '॓']


### Test Dataset
**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 8:1:8  
Loss duration : 0:0:41  
Remaining duration : 8:0:26  
Drop by percentage: 0.14  

Total audio files : 5059  
Files that would be drop: 6  
Remaining audio files : 5053  
Drop by percentage: 0.12  
**************************************************
Non hindi tokes : ['ऍ', 'ऽ', '॓']

In [10]:
# valid_transcription_data=load_transcription(dataset_path/"valid"/"transcription_n2w.txt")
# get_transcription_info(valid_transcription_data,dataset_path,save_path,"dev")
get_stats(save_path/"dev_non_hindi_transcripts.json")

**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 5:0:33
Loss duration : 0:0:14
Remaining duration : 5:0:18
Drop by percentage: 0.08

Total audio files : 3229
Files that would be drop: 2
Remaining audio files : 3227
Drop by percentage: 0.06
**************************************************
Non hindi tokes : ['ऍ', 'ॆ']


### Dev Dataset
**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 5:0:33  
Loss duration : 0:0:14  
Remaining duration : 5:0:18  
Drop by percentage: 0.08  

Total audio files : 3229  
Files that would be drop: 2  
Remaining audio files : 3227  
Drop by percentage: 0.06  
**************************************************
Non hindi tokes : ['ऍ', 'ॆ']

In [11]:
train_transcription_data=load_transcription(dataset_path/"train"/"transcription_n2w.txt")
get_transcription_info(train_transcription_data,dataset_path,save_path,"train")
get_stats(save_path/"train_non_hindi_transcripts.json")

**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 137:19:4
Loss duration : 0:5:10
Remaining duration : 137:13:54
Drop by percentage: 0.06

Total audio files : 93024
Files that would be drop: 46
Remaining audio files : 92978
Drop by percentage: 0.05
**************************************************
Non hindi tokes : ['ऌ', 'ऍ', 'ऎ', 'ऺ', 'ॆ', 'ॊ', 'ॐ', '॓', 'ॠ', 'ॡ', 'ॢ', '॰']


### Train Dataset
**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 137:19:4  
Loss duration : 0:5:10  
Remaining duration : 137:13:54  
Drop by percentage: 0.06  

Total audio files : 93024  
Files that would be drop: 46  
Remaining audio files : 92978  
Drop by percentage: 0.05  
**************************************************
Non hindi tokes : ['ऌ', 'ऍ', 'ऎ', 'ऺ', 'ॆ', 'ॊ', 'ॐ', '॓', 'ॠ', 'ॡ', 'ॢ', '॰']

In [12]:
get_total_stats(save_path)

**************************************************
Loss of audio data for total dataset:
**************************************************
Total duration : 150:20:45
Loss duration : 0:6:6
Remaining duration : 150:14:39
Drop by percentage: 0.07

Total audio files : 101312
Files that would be drop: 54
Remaining audio files : 101258
Drop by percentage: 0.05


### Total Dataset
**************************************************
Loss of audio data for total dataset:
**************************************************
Total duration : 150:20:45  
Loss duration : 0:6:6  
Remaining duration : 150:14:39  
Drop by percentage: 0.07  

Total audio files : 101312  
Files that would be drop: 54  
Remaining audio files : 101258  
Drop by percentage: 0.05  

## Kathbath Clean
test: 1929  
train: 93024  
valid: 3229  
test_known: 3130  
total: 1_01_312  

## Kathbath Hard
test: 816  
test_known: 3034  
total: 3850